libnd4j/include/ops/declarable/platform/cudnn/ctcloss.cu

```diff
@@ -189,14 +192,18 @@ namespace platforms {
         auto outputGrads = OUTPUT_VARIABLE(0);
         int blankIndex = INT_ARG(0);
 
-        auto dTypeInput = logitInput->dataType();
-        auto intType = targetLabelLengths->dataType();
-        auto dTypeOutput = outputGrads->dataType();
+        Requirements req("CUDNN CTC_LOSS_GRAD OP");
```

Comment:
For performance reasons, I would imagine us wanting all of this behind an if debug. Do you mind implementing this?

Reply:
those checks are mandatory. it will just print additional info when debug is enabled ( which may affect performance unnoticeably little as it combine strings to output meaningful messages).

Reply:
I wrote it to replace not only platform checks with meaningful messages but also REQUIRE_TRUE macros (which throws invalid op error and log the error detail). it can do the same. I could even make throwing meaningful errors as well instead of invalid op.
https://godbolt.org/z/M6b3hdnhK
- it is implicitly bool so it can benefit from short circuit &&. And also work inside if statements and return bool expressions
- it can accept lambdas to lazily evaluate values or expressions as well
maybe it will not be as fast as REQUIRE_TRUE macros on some compilers but those checks are not used in inner loops to worry about